oracle/tests/test_unit.py

```diff
@@ -1,12 +1,176 @@
 # (C) Datadog, Inc. 2018-present
 # All rights reserved
-# Licensed under a 3 BSD style license (see LICENSE)
+# Licensed under a 3-clause BSD style license (see LICENSE)
 import copy
 
+try:
+    from contextlib import ExitStack
+except ImportError:
+    from contextlib2 import ExitStack
+
 import mock
 
 from datadog_checks.base.utils.db import Query, QueryManager
-from datadog_checks.oracle import queries
+from datadog_checks.oracle import Oracle, queries
+
+from .common import CHECK_NAME, mock_bad_executor
+
+
+def test__get_connection_instant_client(check, dd_run_check, aggregator):
+    """
+    Test the _get_connection method using the instant client
+    """
+    check.use_oracle_client = True
+    con = mock.MagicMock()
+    expected_tags = ['server:localhost:1521', 'optional:tag1']
+    with mock.patch('datadog_checks.oracle.oracle.cx_Oracle') as cx:
+        cx.connect.return_value = con
+        dd_run_check(check)
+        assert check._cached_connection == con
+        cx.connect.assert_called_with(user='system', password='oracle', dsn=check._get_dsn())
+        aggregator.assert_service_check("oracle.can_connect", check.OK, count=1, tags=expected_tags)
+        aggregator.assert_service_check("oracle.can_query", check.OK, count=1, tags=expected_tags)
+
+
+def test__get_connection_instant_client_query_fail(check, dd_run_check, aggregator):
+    """
+    Test the _get_connection method using the oracle client and unsuccessfully query DB
+    """
+    check.use_oracle_client = True
+    con = mock.MagicMock()
+    expected_tags = ['server:localhost:1521', 'optional:tag1']
+
+    check._query_manager.executor = mock_bad_executor()
+
+    with mock.patch('datadog_checks.oracle.oracle.cx_Oracle') as cx:
+        cx.connect.return_value = con
+        dd_run_check(check)
+        aggregator.assert_service_check("oracle.can_connect", check.OK, count=1, tags=expected_tags)
+        aggregator.assert_service_check("oracle.can_query", check.CRITICAL, count=1, tags=expected_tags)
+
+
+def test__get_connection_instant_client_server_incorrect_formatting(instance, dd_run_check, aggregator):
```

Comment:
@sarah-witt This is the same test that you added in https://github.com/DataDog/integrations-core/pull/9971, just renamed and using aggregator to assert service check.